### PR TITLE
Turn TrackTrigger on.

### PR DESCRIPTION
**Pull request by rekovic**
#### PR description:

Turn L1TrackTrigger on

### CODE REVIEW

**Comment by rekovic** [comment]
please test

File: Configuration/StandardSequences/python/L1TrackTrigger_cff.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from SimTracker.TrackTriggerAssociation.TrackTriggerAssociator_cff import *
 from L1Trigger.TrackerDTC.ProducerED_cff import *
 
-#L1TrackTrigger=cms.Sequence(TrackTriggerClustersStubs*TrackTriggerAssociatorClustersStubs*TrackTriggerTTTracks*TrackTriggerAssociatorTracks)
-L1TrackTrigger=cms.Sequence(TrackTriggerClustersStubs*TrackTriggerAssociatorClustersStubs*TrackerDTCProducer)
+L1TrackTrigger=cms.Sequence(TrackTriggerClustersStubs*TrackTriggerAssociatorClustersStubs*TrackerDTCProducer*TrackTriggerTTTracks*TrackTriggerAssociatorTracks)
```
Comment (silviodonato) [inline Configuration/StandardSequences/python/L1TrackTrigger_cff.py:7]: I don't see any definition of `TrackTriggerTTTracks` in CMSSW

Comment (mtosi) [reply Configuration/StandardSequences/python/L1TrackTrigger_cff.py:7]: sorry for jumping in, but may I ask the reason of the `TrackTriggerAssociatorTracks` in the main L1TrackTrigger sequence ?
it is calling https://github.com/cms-sw/cmssw/blob/6b9ccb57dee572818384626c2db944f300b1fd15/SimTracker/TrackTriggerAssociation/python/TTTrackAssociation_cfi.py

thanks

Comment (rekovic) [reply Configuration/StandardSequences/python/L1TrackTrigger_cff.py:7]: right....
we got to use these  https://github.com/cms-sw/cmssw/blob/master/L1Trigger/TrackFindingTracklet/python/Tracklet_cfi.py#L3

**Comment by rekovic** [comment]
please test abort

File: Configuration/StandardSequences/python/L1TrackTrigger_cff.py (hunk 1 above)
Comment (rekovic) [reply Configuration/StandardSequences/python/L1TrackTrigger_cff.py:7]: my previous comment was to @silviodonato

Comment (rekovic) [reply Configuration/StandardSequences/python/L1TrackTrigger_cff.py:7]: @mtosi 
This is what we have ran in the configuration given to us by TrackTrigger community, 
https://github.com/cms-l1t-offline/cmssw/blob/phase2-l1t-integration-CMSSW_11_1_0_pre6/L1Trigger/L1TCommon/test/reprocess_L1TrackTrigger_L1T.py#L178

If someone from the TrackTrigger community would prefer to make a PR putting the TrackTrigger in the sequence in a preferred way, that would be helpful.  (attn @skinnari )

**Comment by rekovic** [comment]
please abort test

**Comment by rekovic** [comment]
please test

File: Configuration/StandardSequences/python/L1TrackTrigger_cff.py (hunk 1 above)
Comment (rekovic) [reply Configuration/StandardSequences/python/L1TrackTrigger_cff.py]: The commit 2e633c1 should properly addresses the questions discussed above.

**Comment by rekovic** [comment]
Why are there compilation errors reported, if the only change here is the change in sequence.

The errors are from  seem to have to do with L1TTrackMatch package which is in a different PR https://github.com/cms-sw/cmssw/pull/30363
but perhaps has to do with [comment](https://github.com/cms-sw/cmssw/pull/30363/#issuecomment-649534871)

**Comment by rekovic** [comment]
@smuzaffar @silviodonato 
Any idea?

**Comment by silviodonato** [comment]
The TBB error is not due to this PR (see https://github.com/cms-sw/cmsdist/pull/5961#issuecomment-646631169)

**Comment by silviodonato** [comment]
abort

**Comment by silviodonato** [comment]
https://cmssdt.cern.ch/jenkins/job/ib-run-pr-tests/7421/console was already stopped

**Comment by silviodonato** [comment]
test parameters
- pull_request = #30245

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
Let's see if #30245 fixes this error in wf 20034.0, 20434.0, 23234.0, 21234.0 ( cc @skinnari ) 
```
----- Begin Fatal Exception 26-Jun-2020 00:32:53 CEST-----------------------
An exception of category 'BadConfig' occurred while
   [0] Processing  stream begin Run run: 1 stream: 0
   [1] Calling method for module L1FPGATrackProducer/'TTTracksFromExtendedTrackletEmulation'
Exception Message:
/data/cmsbld/jenkins/workspace/build-any-ib/w/tmp/BUILDROOT/a9dd216f82a5b090fd71d59cb74fc34c/opt/cmssw/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-06-24-2300/src/L1Trigger/TrackFindingTracklet/src/MemoryBase.cc 27 name = TPROJ_L2L3A_L1PHIA subname = PH 0 0
----- End Fatal Exception -------------------------------------------------
```

**Comment by silviodonato** [comment]
urgent

**Comment by silviodonato** [comment]
assign upgrade

**Comment by rekovic** [comment]
attn @mtosi @skinnari

**Comment by rekovic** [comment]
@silviodonato 
I don't expect the inclusion of #30245 https://github.com/cms-sw/cmssw/pull/30362#issuecomment-650093044 to fix the failing workflows here, but it is indeed a good idea to test it in conjunction.

File: Configuration/StandardSequences/python/L1TrackTrigger_cff.py [hunk 2]
```diff
@@ -3,8 +3,10 @@
 from L1Trigger.TrackTrigger.TrackTrigger_cff import *
 from SimTracker.TrackTriggerAssociation.TrackTriggerAssociator_cff import *
 from L1Trigger.TrackerDTC.ProducerED_cff import *
+from L1Trigger.TrackFindingTracklet.L1HybridEmulationTracks_cff import *
+
+L1TrackTrigger=cms.Sequence(TrackTriggerClustersStubs*TrackTriggerAssociatorClustersStubs*TrackerDTCProducer*L1PromptExtendedHybridTracksWithAssociators)
+#L1TrackTrigger=cms.Sequence(TrackTriggerClustersStubs*TrackTriggerAssociatorClustersStubs*TrackerDTCProducer*L1PromptExtendedHybridTracks)
```
Comment (kpedro88) [inline Configuration/StandardSequences/python/L1TrackTrigger_cff.py:9]: delete commented-out code

**Comment by rekovic** [comment]
Not sure what the status of the test is here, there is no report, and I don't see any aborting command.  Restarting the tests.

**Comment by rekovic** [comment]
please test

**Comment by silviodonato** [comment]
Please note that after having included #30245 the error is only in 20034.0. The errors in wf 20434.0 21234.0 23234.0 have been fixed.

The remaining error is
```
29-Jun-2020 08:01:28 CEST  Initiating request to open file file:step1.root
29-Jun-2020 08:01:29 CEST  Successfully opened file file:step1.root
create data set info Default
create data set info Default
%MSG-w ConfigurationNotSupported:  AfterModStreamBeginRun  29-Jun-2020 08:02:44 CEST Run: 1 Stream: 0
Geometry Configuration Geometry/TrackerCommonData/data/PhaseII/TiltedTracker404/tracker.xml is not supported. 
%MSG
Begin processing the 1st record. Run 1, Event 1, LumiSection 1 on stream 0 at 29-Jun-2020 08:02:45.924 CEST
----- Begin Fatal Exception 29-Jun-2020 08:03:00 CEST-----------------------
An exception of category 'BadConfig' occurred while
   [0] Processing  Event run: 1 lumi: 1 event: 1 stream: 0
   [1] Running path 'L1TrackTrigger_step'
   [2] Calling method for module L1FPGATrackProducer/'TTTracksFromExtendedTrackletEmulation'
Exception Message:
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-06-28-0000/src/L1Trigger/TrackFindingTracklet/src/Stub.cc 145 didn't find rDSS value! r = 65.1896 Check that correct geometry is used!
----- End Fatal Exception -------------------------------------------------
```

Workflow 20034.0 uses the oldest geometry
20034.0 -> D35
20434.0 -> D41
21234.0 -> D44
23234.0 -> D49

D35 is the MTD TDR baseline, D41 is the L1T TDR baseline, and D49 is the HLT TDR baseline [Configuration/Geometry](https://github.com/cms-sw/cmssw/tree/master/Configuration/Geometry).

**Comment by rekovic** [comment]
@silviodonato

> D35 is the MTD TDR baseline, D41 is the L1T TDR baseline, and D49 is the HLT TDR baseline Configuration/Geometry.


Yes, that is correct.   
The latest Hybrid Tracklet code does not run D35 out of box with simple change in configuration.  Unfortunately, a non-configurable change in code is needed. 
https://github.com/cms-l1t-offline/cmssw/blob/phase2-l1t-integration-CMSSW_11_1_0_pre6/L1Trigger/L1TCommon/test/reprocess_L1TrackTrigger_L1T.py#L18
And this is why the test 20034.0 fails. 

This situation was not an obstacle for us and HLT in the studies done currently, since we processed L1T TDR samples (D41) and HLT TDR samples (D49).  However, this is not a sustainable solution. 
We either (a) request the TrackTrigger code to be made compatible with D35, or (b) we remove "L1TrackTrigger" step from workflow that includes geometry D35, i.e. from 20034.0.

**Comment by rekovic** [comment]
Preference is for (a) above.  
Attn @skinnari.

**Comment by silviodonato** [comment]
In https://github.com/cms-l1t-offline/cmssw/pull/847 I suggest a possible workaround to disable L1TrackTrigger in D35

**Comment by silviodonato** [comment]
moved to #30479

**Pull request closed without merging**